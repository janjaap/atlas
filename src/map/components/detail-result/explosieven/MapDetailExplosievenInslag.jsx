import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultDateItem from '../MapDetailResultDateItem';
import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailExplosievenInslag = ({ panoUrl, inslag, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={inslag.label}
    title="Inslag"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultDateItem
        label="Datum van inslag"
        date={inslag.date}
      />
      <MapDetailResultItem
        label="Soort handeling"
        value={inslag.type}
      />
      <MapDetailResultItem
        label="Bron"
        value={inslag.source}
      />
      <MapDetailResultItem
        label="Opmerkingen"
        value={inslag.remarks}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailExplosievenInslag.propTypes = {
  inslag: PropTypes.shape({
    date: PropTypes.instanceOf(Date),
    label: PropTypes.string,
    remarks: PropTypes.string,
    source: PropTypes.string,
    type: PropTypes.string
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailExplosievenInslag;
