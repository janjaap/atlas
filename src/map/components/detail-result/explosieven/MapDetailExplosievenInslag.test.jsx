import React from 'react';
import { shallow } from 'enzyme';

import MapDetailExplosievenInslag from './MapDetailExplosievenInslag';

describe('MapDetailExplosievenInslag', () => {
  it('should render everything', () => {
    const inslag = {
      date: new Date('1980-08-19'),
      label: 'Inslag label',
      remarks: 'Inslag remarks',
      source: 'Inslag source',
      type: 'Inslag type'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailExplosievenInslag
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        inslag={inslag}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
