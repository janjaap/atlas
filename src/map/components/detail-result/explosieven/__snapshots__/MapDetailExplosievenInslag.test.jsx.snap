// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailExplosievenInslag should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Inslag label"
  title="Inslag"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultDateItem
      date={1980-08-19T00:00:00.000Z}
      label="Datum van inslag"
    />
    <MapDetailResultItem
      label="Soort handeling"
      value="Inslag type"
    />
    <MapDetailResultItem
      label="Bron"
      value="Inslag source"
    />
    <MapDetailResultItem
      label="Opmerkingen"
      value="Inslag remarks"
    />
  </ul>
</MapDetailResultWrapper>
`;
